pkg/route/validation/validation_test.go

```diff
@@ -1200,6 +1687,119 @@ func TestValidateRouteUpdate(t *testing.T) {
+		{description: "quoted value", input: `%{+Q}foo`, validInput: false},
+		{description: "hdr with empty field", input: `%[XYZ.hdr()]`, validInput: false},
+		{description: "hdr with percent field", input: `%[XYZ.hdr(%)]`, validInput: false},
+		{description: "hdr with known field", input: `%[XYZ.hdr(Host)]`, validInput: true},
+		{description: "hdr with syntax error", input: `%[XYZ.hdr(Host]`, validInput: false},
+
+		{description: "hdr with valid X-XSS-Protection value", input: `1;mode=block`, validInput: true},
+		{description: "hdr with valid Content-Type value", input: `text/plain,text/html`, validInput: true},
+
+		{description: "incomplete expression", input: `%[req`, validInput: false},
+		{description: "quoted field", input: `%[XYZ.hdr(%{+Q}Host)]`, validInput: false},
+		{description: "value with conditional expression", input: `%[XYZ.hdr(Host)] if foo`, validInput: true},
+		{description: "value with what looks like a conditional expression", input: `%[XYZ.hdr(Host)]\ if\ foo`, validInput: true},
+		{description: "unsupported fetcher and converter", input: `%[date(3600),http_date]`, validInput: false},
+		{description: "not allowed sample fetchers", input: `%[foo,lower]`, validInput: false},
+		{description: "not allowed converters", input: `%[req.hdr(host),foo]`, validInput: false},
+		{description: "missing parentheses or braces", input: `%{Q[req.hdr(host)]`, validInput: false},
+		{description: "missing parentheses or braces", input: `%Q}[req.hdr(host)]`, validInput: false},
+		{description: "missing parentheses or braces", input: `%{{Q}[req.hdr(host)]`, validInput: false},
+		{description: "missing parentheses or braces", input: `%[req.hdr(host)`, validInput: false},
+		{description: "missing parentheses or braces", input: `%req.hdr(host)]`, validInput: false},
+		{description: "missing parentheses or braces", input: `%[req.hdrhost)]`, validInput: false},
+		{description: "missing parentheses or braces", input: `%[req.hdr(host]`, validInput: false},
+		{description: "missing parentheses or braces", input: `%[req.hdr(host`, validInput: false},
+		{description: "missing parentheses or braces", input: `%{req.hdr(host)}`, validInput: false},
+		{description: "parameters for a sample fetcher that doesn't take parameters", input: `%[ssl_c_der(host)]`, validInput: false},
+		{description: "dangerous sample fetchers and converters", input: `%[env(FOO)]`, validInput: false},
+		{description: "dangerous sample fetchers and converters", input: `%[req.hdr(host),debug()]`, validInput: false},
+		{description: "extra comma", input: `%[req.hdr(host),,lower]`, validInput: false},
+
+		// CR and LF are not allowed in header value as per RFC https://datatracker.ietf.org/doc/html/rfc7230#section-3.2.4
+		{description: "carriage return", input: "\r", validInput: false},
+		{description: "CRLF", input: "\r\n", validInput: false},
+
+		{description: "environment variable with a bracket missing", input: `${NET_COOLOCP_HOSTPRIMARY`, validInput: true},
```

Comment:
Why is this valid, because it is just a string?  Not really an environment variable?

Reply:
Why do env variables not require a terminating curly brace?

Reply:
It is considered as a string quoted value. User is responsible to provide a correct value. We escape and quote this value so such values don't cause haproxy to crash.

Reply:
Based on this discussion, it makes sense to me to add a comment:
```go
		// HAProxy does not interpret ${} syntax, so it is safe to have ${ in the value; it will be taken
		// literally as part of the header value.
		{description: "environment variable with a bracket missing", input: `${NET_COOLOCP_HOSTPRIMARY`, validInput: true},
```

Reply:
ok added this doc